From oliver@neukum.org  Fri Oct  9 11:43:28 2009
From: Oliver Neukum <oliver@neukum.org>
Date: Thu, 1 Oct 2009 15:01:17 +0200
Subject: USB: small fix in error case of suspend in generic usbserial code
To: "Greg Kroah-Hartman" <gregkh@suse.de>
Message-ID: <200910011501.17797.oliver@neukum.org>
Content-Disposition: inline

usb:usbserial: fix flags in error case of suspension

suspended flag must be reset in error case

Signed-off-by: Oliver Neukum <oliver@neukum.org>
Cc: stable <stable@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/usb-serial.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/usb/serial/usb-serial.c
+++ b/drivers/usb/serial/usb-serial.c
@@ -1168,8 +1168,10 @@ int usb_serial_suspend(struct usb_interf
 
 	if (serial->type->suspend) {
 		r = serial->type->suspend(serial, message);
-		if (r < 0)
+		if (r < 0) {
+			serial->suspending = 0;
 			goto err_out;
+		}
 	}
 
 	for (i = 0; i < serial->num_ports; ++i) {
